### ISSUE TITLE
MongoDB download not working for Ubuntu 24.04

### ISSUE THREAD

**Issue by Euquimides** [craftercms/craftercms#7385]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [ ] The issue is in the latest released 4.1.x

### Describe the issue

MongoDB download (which is used for Social and Profile bundles) isn't working for users using Ubuntu 24.04. 

The working download links seems to be `https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2404-6.0.5.tgz`. 

### Steps to reproduce

Steps:

While having Ubuntu 24.04:

1. Clone Crafter.
2. Run `./gradlew clean build deploy -PquietBuild=false -PrefreshEnv=true -PoverwriteArtifacts=true -PsocialRequired=true -PprofileRequired=true`
3. See the issue

### Relevant log output

```shell
> Task :splash

 ██████╗ ██████╗   █████╗  ███████╗ ████████╗ ███████╗ ██████╗   ██████╗ ███╗   ███╗ ███████╗
██╔════╝ ██╔══██╗ ██╔══██╗ ██╔════╝ ╚══██╔══╝ ██╔════╝ ██╔══██╗ ██╔════╝ ████╗ ████║ ██╔════╝
██║      ██████╔╝ ███████║ █████╗      ██║    █████╗   ██████╔╝ ██║      ██╔████╔██║ ███████╗
██║      ██╔══██╗ ██╔══██║ ██╔══╝      ██║    ██╔══╝   ██╔══██╗ ██║      ██║╚██╔╝██║ ╚════██║
╚██████╗ ██║  ██║ ██║  ██║ ██║         ██║    ███████╗ ██║  ██║ ╚██████╗ ██║ ╚═╝ ██║ ███████║
 ╚═════╝ ╚═╝  ╚═╝ ╚═╝  ╚═╝ ╚═╝         ╚═╝    ╚══════╝ ╚═╝  ╚═╝  ╚═════╝ ╚═╝     ╚═╝ ╚══════╝


> Task :download FAILED
Download https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2404-6.0.5.tgz

FAILURE: Build failed with an exception.

* Where:
Script '/home/juanvasquez/crafter/craftercms-copy/craftercms/downloads.gradle' line: 117

* What went wrong:
Execution failed for task ':download'.
> de.undercouch.gradle.tasks.download.org.apache.hc.client5.http.ClientProtocolException: Forbidden (HTTP status code: 403, URL: https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2404-6.0.5.tgz)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 606ms
3 actionable tasks: 3 executed
```

**Comment by phuongnq** [comment]
Need to upgrade to Mongo v8.
Refer to platform support matrix: https://www.mongodb.com/docs/v6.0/administration/production-notes/#platform-support-matrix

**Comment by phuongnq** [comment]
Note: In v8, the option `--journal` is removed as it is enabled by default: https://www.mongodb.com/docs/manual/core/journaling/
> Starting in MongoDB 6.1, journaling is always enabled. As a result, MongoDB removes the storage.journal.enabled option and the corresponding --journal and --nojournal command-line options.

**Comment by sumerjabri** [comment]
Upgrading MongoDB https://www.mongodb.com/docs/manual/release-notes/8.0-upgrade-standalone/#upgrade-version-path

> To upgrade an existing MongoDB deployment to 8.0, you must be running a 7.0-series release.

> To upgrade from a version earlier than the 7.0-series, you must successively upgrade major releases until you have upgraded to 7.0-series. For example, if you are running a 6.0-series, you must [upgrade first to 7.0](https://www.mongodb.com/docs/manual/release-notes/7.0/#std-label-7.0-upgrade) before you can upgrade to 8.0.

**Comment by Euquimides** [comment]
Hi! Updated and rechecked this issue. Fix verified. Thank you! Closing this ticket.

**Comment by sumerjabri** [comment]
@Euquimides please validate:
* Profile
* Social
* Upgrade from 4.1 to 4.2 of Profile/Social

**Comment by Euquimides** [comment]
Hi @sumerjabri ! I already tested this ticket for Profile and Social, only the upgrade form 4.1 -> 4.2. of Profile/Social is left. What I am noticing is that this issue is present in `v4.1.9` so we might need to fix this version so that the correct MongoDB version is downloaded according to Crafter version and the user's OS version, else MongoDB will not be downloaded automatically for versions below 4.2 in Ubuntu 24.04. 

```
➜  craftercms git:(support/4.1.x) ./gradlew clean build deploy -PquietBuild=false -PrefreshEnv=true -PoverwriteArtifacts=true -PsocialRequired=true -PprofileRequired=true

> Task :splash

 ██████╗ ██████╗   █████╗  ███████╗ ████████╗ ███████╗ ██████╗   ██████╗ ███╗   ███╗ ███████╗
██╔════╝ ██╔══██╗ ██╔══██╗ ██╔════╝ ╚══██╔══╝ ██╔════╝ ██╔══██╗ ██╔════╝ ████╗ ████║ ██╔════╝
██║      ██████╔╝ ███████║ █████╗      ██║    █████╗   ██████╔╝ ██║      ██╔████╔██║ ███████╗
██║      ██╔══██╗ ██╔══██║ ██╔══╝      ██║    ██╔══╝   ██╔══██╗ ██║      ██║╚██╔╝██║ ╚════██║
╚██████╗ ██║  ██║ ██║  ██║ ██║         ██║    ███████╗ ██║  ██║ ╚██████╗ ██║ ╚═╝ ██║ ███████║
 ╚═════╝ ╚═╝  ╚═╝ ╚═╝  ╚═╝ ╚═╝         ╚═╝    ╚══════╝ ╚═╝  ╚═╝  ╚═════╝ ╚═╝     ╚═╝ ╚══════╝


> Task :download FAILED
Download https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2404-6.0.5.tgz

FAILURE: Build failed with an exception.

* Where:
Script '/home/juanvasquez/crafter/craftersupport41/craftercms/downloads.gradle' line: 117

* What went wrong:
Execution failed for task ':download'.
> de.undercouch.gradle.tasks.download.org.apache.hc.client5.http.ClientProtocolException: Forbidden (HTTP status code: 403, URL: https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2404-6.0.5.tgz)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
3 actionable tasks: 3 executed
```

**Comment by sumerjabri** [comment]
@Euquimides we can't upgrade a major dependency (MongoDB from 6 -> 8) in a patch release (4.1.9).

Please report back the results of testing the upgrades when you're done.

**Comment by Euquimides** [comment]
Hi! Tested Social and Profile. Also made an upgrade of MongoDB from:

1. 4.1.x -> 4.2.0 (from MongoDB 7 to 8)
2. 4.0.x -> 4.2.0 (from MongoDB 4 -> 4.2 -> 5 -> 6 -> 7 -> 8)

Social and Profile modules seems to work fine after upgrade. Fix verified for v4.2.0. Thank you! Closing this ticket.
